### PR TITLE
feat(collapsible): functional component

### PR DESCRIPTION
**Pull request by pochretien**
Refactor collapsible to use functional component.
Keep the old not connected component class to refer to the connected.
update uts to remove the usage of mount

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/src/components/collapsible/tests/CollapsibleConnected.spec.tsx [hunk 1]
```diff
@@ -1,140 +1,76 @@
...
-            </Provider>,
-            {attachTo: document.getElementById('App')}
+    let store: ReturnType<typeof getStoreMock>;
+    let wrapper: ReactWrapper<CollapsibleOwnProps>;
+    const collapsibleProps = collapsiblePossibleProps[0];
+
+    const mountComponentWithProps = (props: Partial<CollapsibleOwnProps> = {}, expanded: boolean = true) => {
+        store = getStoreMock({
+            collapsibles: [{id: collapsibleProps.id, expanded}],
+        } as any);
```
Comment (rosalie-liu) [inline packages/react-vapor/src/components/collapsible/tests/CollapsibleConnected.spec.tsx:19]: Does it work with `} as IReactVaporState`?

File: packages/react-vapor/src/components/collapsible/tests/CollapsibleContainerConnected.spec.tsx [hunk 2]
```diff
@@ -1,77 +1,69 @@
...
-    const mountComponentWithProps = (props: ICollapsibleContainerOwnProps) => {
-        return mount(
-            <Provider store={store}>
-                <CollapsibleContainerConnected {...props}>dummy children</CollapsibleContainerConnected>
-            </Provider>,
-            {attachTo: document.getElementById('App')}
+    const mountComponentWithProps = (props: Partial<ICollapsibleContainerOwnProps> = {}, expanded: boolean = true) => {
+        store = getStoreMock({
+            collapsibles: [{id: collapsibleProps.id, expanded}],
+        } as any);
```
Comment (rosalie-liu) [inline packages/react-vapor/src/components/collapsible/tests/CollapsibleContainerConnected.spec.tsx:24]: same here

**Comment by rosalie-liu** [comment]
I'm thinking about an UX improvement which might be interesting for `CollapsibleContainerConnected`. When we implement a form into a `CollapsibleContainerConnected`, the title could be displayed differently if the form is invalid, as what we have in Source Authentication.
https://bitbucket.org/coveord/admin-ui/src/3fba057f79192eeb86fd5102224c23247067b5d9/packages/v2/src/sources/sources/components/complex-authentication/ComplexAuthentication.tsx#lines-45
image
We could probably add a prop `hasError` or `isValid`, when the form is not valid, the title of `CollapsibleContainerConnected` could be changed to red and add a warning icon on the left side.
Btw, this may be a little out of scope for this PR, we could probably create a small user story of UX improvement for that, if it can be reusable in the future.

File: packages/react-vapor/src/components/collapsible/CollapsibleConnected.tsx [hunk 3]
```diff
@@ -22,15 +29,54 @@ const mapStateToProps = (state: IReactVaporState, ownProps: CollapsibleOwnProps)
...
-const mapDispatchToProps = (dispatch: IDispatch, ownProps: CollapsibleOwnProps): CollapsibleDispatchProps => ({
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: CollapsibleOwnProps) => ({
     onMount: () => dispatch(addCollapsible(ownProps.id, !!ownProps.expandedOnMount)),
     onUnmount: () => dispatch(removeCollapsible(ownProps.id)),
     onToggleExpandedState: (currentExpandedState: boolean) =>
         dispatch(setCollapsibleExpanded(ownProps.id, !currentExpandedState)),
 });
 
-export const CollapsibleConnected: React.ComponentClass<CollapsibleProps> = connect(
+export const COLLAPSIBLE_TIME = 150;
```
Comment (gdostie) [inline packages/react-vapor/src/components/collapsible/CollapsibleConnected.tsx:39]: We could add this one the `Defaults`
```suggestion
export const COLLAPSIBLE_EFFECT_DURATION = 150;

```

File: packages/react-vapor/src/components/collapsible/CollapsibleContainerConnected.tsx [hunk 4]
```diff
@@ -3,24 +3,110 @@ import {findWhere} from 'underscore';
...
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: ICollapsibleContainerOwnProps) => {
     const collapsibleState = findWhere(state.collapsibles, {id: ownProps.id});
 
     return {expanded: collapsibleState && collapsibleState.expanded};
 };
 
-export const CollapsibleContainerConnected: React.ComponentClass<ICollapsibleContainerProps> = connect(
+const getCollapsibleHeaderIcon = ({
```
Comment (gdostie) [inline packages/react-vapor/src/components/collapsible/CollapsibleContainerConnected.tsx:35]: Convert to function component?
```suggestion
const CollapsibleHeaderIcon: React.FunctionComponent<{
    informationTooltip: ITooltipProps;
    informationUrl: string;
}> = ({informationTooltip, informationUrl}) =>
```

**Pull request merged**